src/pybind/mgr/progress/module.py

```diff
@@ -301,7 +313,8 @@ def pg_update(self, raw_pg_stats, pg_ready, log):
         # FIXME: far more fields getting pythonized than we really care about
         # Sanity check to see if there are any missing PGs and to assign
         # empty array and dictionary if there hasn't been any recovery
-        pg_to_state = dict([(p['pgid'], p) for p in raw_pg_stats['pg_stats']]) # type: Dict[str, Any]
+
+        pg_to_state = dict([(p['pgid'], p) for p in raw_pg_stats['pg_stats']])  # type: Dict[str, Any]
```

Comment:
no need to materialize the generator to a list before converting them to a dict. you can just put
```py
pg_to_state = dict((p['pgid'], p) for p in raw_pg_stats['pg_stats'])
```